streams/src/main/java/org/apache/kafka/streams/kstream/internals/StreamsTopologyGraph.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kafka.streams.kstream.internals;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * The builder of the Streams Intermediate DAG used to make
+ * optimizations of a Kafka Streams topology and sub-topologies
+ */
+public abstract class StreamsTopologyGraph {
```

Comment:
Why an abstract class here? Why not an interface and the single implementation? I don't see this adding any value

Reply:
I had an interface initially, I can't remember the rationale for changing, so putting it back to an interface.